satpy/composites/__init__.py

```diff
@@ -1477,6 +1460,44 @@ def _set_data_nans(self, data, mask, attrs):
 
         return data
 
+    def _select_data_bands(self, data_in):
+        """Select data to be composited from input data.
+
+        From input data, select the bands that need to have masking applied.
+        """
+        if 'bands' in data_in.dims:
+            return [data_in.sel(bands=b) for b in data_in['bands'] if b != 'A']
+        if self.mode == "RGBA":
+            return [data_in, data_in, data_in]
```

Comment:
These are the only two lines here that are actually new.  The rest is just refactoring.